text/0008-uniform-random-tip-selection/0008-uniform-random-tip-selection.md

```diff
@@ -1,206 +1,156 @@
+relevant in relation to the recent parts of the Tangle. The current `BMD` for mainnet nodes is 15 milestones, 
+which means that transactions of which their `OTRSI` in relation to the `LSMI` is more than 15, are "below max depth".
+
+### OTRSI / YTRSI example
+Given the blue PoV transaction, the `OTRSI` of it is milestone 1 and `YTRSI` milestone 2.
+Note that here again, the milestones are also `Confirmed Root Transactions`.
+![sdf](images/otrsi_ytrsi.PNG)
+
+### Milestone based tip scoring
+
+The milestone based scoring defines a tip's score by investigating the tip's relation to
+the cone it approves and previous issued  milestones.
+
+A tip can have one of 3 score states:
+* `0`: The tip is lazy and should not be selected.
+* `1`: The tip is somewhat lazy.
+* `2`: The tip is a non-lazy tip.
+
+Definitions:
+* `C1`: Max allowed delta value for the `YTRSI` of a given transaction in relation to the current `LSMI`.
+* `C2`: Max allowed delta value between `OTRSI` of the approvees of a given transaction in relation to the current `LSMI`. 
+* `M`: Max allowed delta value between `OTRSI` of the given transaction in relation to the current `LSMI`.
+`M` is the `below max depth (BMD)` parameter.
+
+Recommended defaults:
+* `C1` = 8 milestones
+* `C2` = 13 milestones
+* `M` = 15 milestones
+
+Scoring Algorithm (pseudo code):
+```
+
+enum Score (
+    LAZY = 0
+    SEMI_LAZY = 1
+    NON_LAZY = 2
+)
+
+const (
+    C1 = 8
+    C2 = 13
+    M = 15
+)
+
+func score(tip Tip) Score {
+    
+    // if the LSMI to YTRSI delta is over C1, then the tip is lazy
+    if (LSMI - YTRSI(tip) > C1) {
+        return Score.LAZY
+    }
+    
+    // if the OTRSI to LSMI delta is over M/below-max-depth, then the tip is lazy
+    if (LSMI - OTRSI(tip) > M) {
+        return Score.LAZY
+    }
+    
+    if (LSMI - ORTSI(tip) > C2) {
+        return Score.SEMI_LAZY
+    }
+
+    return Score.NON_LAZY
+}
+```
+
+### Random Tip-Selection
+
+A node should keep a set of non-lazy tips (score 2).
+Every time a node is asked to select to tips to be approved it will just pick randomly from the set. 
+A node must not execute tip-selection if it is not synchronized.
+
+A tip should not be removed from the tips set immediately after it was selected in `select()`, 
+in order to make it possible for it to be re-selected, which in turn makes the Tangle wider
+and improves synchronization speed. A tip is removed from the tips set if `X` amount of direct
+approvers are reached or if a certain amount of time `T` passed. 
+It is recommended to use `X` = 2 and `T` = 3  but the threshold should be configurable.
+
+### Purpose Of Semi-Lazy Tips
+
+Semi-Lazy tips are not eligible for tip-selection, but the coordinator node may implement a tip selection algorithm
+that confirms semi-lazy tips. Semi-lazy tips will usually be left behind, but parties interested in having them confirmed
+are incentivized to run spammers that will merge them into one thin subtangle. 
```

Comment:
> will merge them into one thin subtangle. 

I find this sentence confusing: I would rephrase it in something like "... incentivized to run spammers that will actively reduce the amount of semi-lazy tips eligible for coordinator's tip selection"